src/sentry/api/serializers/models/team.py

```diff
@@ -93,6 +93,33 @@ def get_member_totals(team_list: Sequence[Team], user: User) -> Mapping[str, int
     return {item["id"]: item["member_count"] for item in query}
 
 
+def get_member_orgs_and_roles(org_ids: Set[int], user_id: int) -> Mapping[int, Sequence[str]]:
+    org_members = OrganizationMember.objects.filter(
+        user_id=user_id, organization__in=org_ids
+    ).values_list("organization_id", "id", "role")
```

Comment:
I had to look twice at this to ensure it wouldn't cause redundant database reads. But I [checked the docs on caching](https://docs.djangoproject.com/en/4.1/topics/db/queries/#caching-and-querysets) and this seems to be the correct way to do it. 😅